src/test/java/core/basesyntax/service/impl/FileWriterCsvImplTest.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    @Test
+    void write_reportIsEmpty_NotOk() {
+        List<String> emptyReport = new ArrayList<>();
+        Throwable exception = assertThrows(RuntimeException.class,
+                () -> FILE_WRITER_CSV.write(fileName, emptyReport));
+        assertEquals("Report is empty", exception.getMessage());
+    }
+
+    @Test
+    void write_fileNameIsEmpty_NotOk() {
+        String emptyFileName = "";
+        Throwable exception = assertThrows(RuntimeException.class,
+                () -> FILE_WRITER_CSV.write(emptyFileName, report));
+        assertEquals("File name is empty", exception.getMessage());
+    }
+
+    @Test
+    void write_notCsvFileExtension_NotOk() {
+        List<String> wrongFileFormatList = new ArrayList<>();
+        wrongFileFormatList.add("src/test/resources/writer/reportCsv.xml");
+        wrongFileFormatList.add("src/test/resources/writer/reportCsv.txt");
+        wrongFileFormatList.add("src/test/resources/writer/reportCsv.doc");
+        wrongFileFormatList.add("src/test/resources/writer/reportCsv.jpg");
+        wrongFileFormatList.add("src/test/resources/writer/reportCsv.java");
+        wrongFileFormatList.add("src/test/resources/writer/reportCsv.pdf");
+        for (String path : wrongFileFormatList) {
+            Throwable exception = assertThrows(RuntimeException.class,
+                    () -> FILE_WRITER_CSV.write(path, report));
+            assertEquals("The file extension is different from .csv: "
+                    + path, exception.getMessage());
+        }
+    }
+
+    @Test
+    void write_nullValueInReport_NotOk() {
+        List<String> listWithNull1 = new ArrayList<>();
+        listWithNull1.add(null);
+        listWithNull1.add(null);
+        listWithNull1.add(null);
+        listWithNull1.add(null);
+        List<String> listWithNull2 = new ArrayList<>();
+        listWithNull2.add("line");
+        listWithNull2.add("line");
+        listWithNull2.add("line");
+        listWithNull2.add(null);
+        Throwable exception1 = assertThrows(RuntimeException.class,
+                () -> FILE_WRITER_CSV.write(fileName, listWithNull1));
+        assertEquals("Report contains null value", exception1.getMessage());
+        Throwable exception2 = assertThrows(RuntimeException.class,
+                () -> FILE_WRITER_CSV.write(fileName, listWithNull2));
+        assertEquals("Report contains null value", exception2.getMessage());
+    }
+
+    @Test
+    void write_correctParameters_Ok() {
+        List<String> linesFromFile;
+        FILE_WRITER_CSV.write(fileName, report);
+        try {
+            Path path = Path.of(fileName);
+            linesFromFile = Files.readAllLines(path);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from file");
+        }
+        boolean isListEquals = Arrays.equals(report.toArray(), linesFromFile.toArray());
+        assertTrue(isListEquals);
+    }
+
+    @Test
+    void write_fileAlreadyExist_Ok() {
```

Comment:
order of execution is unpredictable, do you rely that the file is created in the previous test?